### PR TITLE
Initial Commit of the TDM gamemode

### PR DESCRIPTION
**Pull request by Hughgent**
## About The Pull Request

This is the initial thoughts on creating a ticket based TDM gamemode.

It is quite incomplete at this point though.

## Why It's Good For The Game

Somewhat of a core feature

## Changelog
:cl: Hughgent
add: TDM Gamemode
/:cl:

### CODE REVIEW

File: code/datums/gamemodes/TDM_tickets.dm [hunk 1]
```diff
@@ -0,0 +1,73 @@
+/datum/game_mode/team_death_match
+	name = "Team Death Match"
+	config_tag = "TDM"
+	required_players = 0 //change to 2 in the future
```
Comment (mattmatterson111) [inline code/datums/gamemodes/TDM_tickets.dm:4]: Warfare has 0 required players. Leaving it at 0 just makes testing easier. If you're going to play a game it's pretty much guaranteed you're going to have _at least_ 2 players.

File: code/datums/gamemodes/TDM_tickets.dm [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	)
+
+//yeah yeah move this to a defines file at some point in the future.
+#define A_TEAM_VICTORY "a_team_victory"
+#define B_TEAM_VICTORY "b_team_victory"
+
+//for when scaling of the round start tickets is important
+///datum/game_mode/team_death_match/proc/scale_tickets()
+//	a_team_tickets = 50 * CONFIG_GET(number/ticket_scale)
+//	b_team_tickets = 50 * CONFIG_GET(number/ticket_scale)
```
Comment (mattmatterson111) [inline code/datums/gamemodes/TDM_tickets.dm:41]: Make sure you uncomment and implement this if you actually plan on using this feature.

File: code/modules/client/preferences.dm [hunk 3]
```diff
@@ -208,6 +208,7 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 	dat += "<a href='?_src_=prefs;preference=randomize_name'>(R)</a>"
 	dat += "<br>"
 	dat += "Always Pick Random Name: <a href='?_src_=prefs;preference=random_name'>[random_name ? "Yes" : "No"]</a>"
+	dat += "<br><b>Citizenship</b>: <a href = '?_src_=prefs;preference=citizenship'>[citizenship]</a><br/>"
```
Comment (mattmatterson111) [inline code/modules/client/preferences.dm:211]: Letting players pick the side their on is a bad idea. I recommend autoabalancing the client to one side or the other side when they connect.

Comment (Hughgent) [reply code/modules/client/preferences.dm:211]: I would imagine this being more of a 'preference' for the player. Autobalance should indeed still happen.

**Comment by TheBimmer231** [comment]
considering the newest sprites made, this'll also need a way to create capture objectives, based on how much time members of a team stand there uncontested, which displays the information somewhere, and reduces the tickets at some rate for the duration that the enemy team has it.
and of course a way to reduce tickets another way.

**Pull request closed without merging**